docs/source/install.md

```diff
@@ -1,9 +1,146 @@
 # Installation
 
-## TODO:
 
-- Document traefik-proxy installation
-- enabling traefik in jupyterhub_config.py
-- what's needed (traefik binary, etcd binary)
-- cover how we talk to traefik (api endpoint, auth)
-- link to toml, etcd
+## Traefik-proxy installation
+
+1. Install **traefik-proxy** through the project’s [Github repository](https://github.com/jupyterhub/traefik-proxy):
+
+    ```
+    $ git clone https://github.com/jupyterhub/traefik-proxy.git
+    $ cd traefik-proxy
+    $ pip install .
+    ```
+
+2. Install **Jupyterhub**:
+    ```
+    $ python -m pip install jupyterhub
+    ```
+
+3. In order to be able to launch JupyterHub with traefik-proxy or run the tests, **traefik** and **etcd** must first be installed and added to your `PATH`.
+   
+   There are two ways you can install traefik and etcd:
+   
+   1. Through traefik-proxy's **install utility**.
+
+      From `traefik-proxy` directory:
+
+      ```
+      $ python -m jupyterhub_traefik_proxy.install --output=/usr/local/bin
+      ```
+     
+      This will install the default versions of traefik and etcd, namely `traefik-1.7.5` and `etcd-3.3.10` to `/usr/local/bin` specified through the `--output` option.
+
+      If no directory is passed to the installer, a *dependencies* directory will be created in the `traefik-proxy` directory. In this case, you **must** add this directory to `PATH`, e.g.
+
+      ```
+      $ export PATH=$PATH:{$PWD}/dependencies
+      ```
+
+      If you want to install other versions of traefik and etcd in a directory of your choice, just specify it to the installer through the following arguments:
+        * `--traefik-version`
+        * `--etcd-version`
+        * `--output`
+
+      Example
+      ```
+      $ python -m jupyterhub_traefik_proxy.install --output=dep \
+               --traefik-version=1.6.6 --etcd-version=3.2.24
+      ```
+
+      If the desired install directory doesn't exist, it will be created by the installer.
+
+    2. From traefik and etcd **release pages**:
```

Comment:
Rather than detailing the installation commands, it might be worth just linking to the projects' own installation instructions:

- traefik https://traefik.io/#easy-to-install
- etcd: https://github.com/etcd-io/etcd/releases